Correccion, refactorizacion y bootstrap - tarea 1

gruvix opened the pull request:
-Se refactorizó la tarea 1
-Se agrega bootstrap a ambas tareas, y se implementa en la tarea 1 (ahora se ve súper bonito)
-Se mejoraron validaciones de la tarea 1

gruvix committed:
add bootstrap

gruvix committed:
add reference to bootstrap in html's

gruvix committed:
add install instructions

gruvix committed:
Renombre funciones

renombre funciones para mejorar la comprension futura

gruvix committed:
refactor click "Siguiente"

reformulado del resultado de clickear siguiente para que sea mas facil distinguir las partes del codigo
Ahora se fija si hay un error proveniente de validarCantidadFamiliares
si no hay error funciona normalmente

gruvix committed:
Revert "Renombre funciones"

This reverts commit 145e6d44d7641a52635a52cfc319e7f2faf62ebf.

gruvix committed:
removi retorno predeterminado de validacion de cantidad

gruvix committed:
renombre validar input  a validarCantidadFamiliares

para que sea mas explícito con lo que hace

gruvix committed:
add bootstrap

agrega los estilos de bootstrap a la mayoria del html, mejorando visualmente el contenido

gruvix committed:
cambia la clase de los inputs de edades

la nueva clase proviene de bootstrap, dejando un estilo mas bonito
la clase anterior se borra de css ya no se usa

gruvix committed:
fix reiniciar nuevamente remueve personas

gruvix committed:
refactor llamado a validarEdad

ahora se define la constante edad antes de validarla en otra funcion

gruvix committed:
add funcion ocultar bloque de errores de edad

gruvix committed:
error cantidad ahora tiene su propio bloque

ya no se muestra adentro del bloque de seleccion de cantidad, sino que aparece como un error aparte

gruvix committed:
add ocultar bloque de inputs de las edades

ahora el bloque de los inputs se oculta y se muestra cuando es necesario

gruvix committed:
fix instrucciones de instalacion erroneas

GonzaloAlt commented on js/tarea1.js:
```diff
@@ -1,36 +1,52 @@
 document.querySelector("#calcular").addEventListener("click", function () {
-    const gente = document.querySelectorAll(".persona")
+    const gente = document.querySelectorAll(".input-group-text")
     if(validarEdades(gente)){
```
Si está `validarEdades(gente)`  valida edades, porque recibe gente? Que devuelve para que sea condición de ese if? creo que sería mejor sacarlo por fuera del if y guardarlo en una variable antes de evaluar la condición, ayuda a la sintaxis y ala comprensión.

GonzaloAlt commented on js/tarea1.js:
```diff
@@ -1,36 +1,52 @@
-    const gente = document.querySelectorAll(".persona")
+    const gente = document.querySelectorAll(".input-group-text")
     if(validarEdades(gente)){
         datos = calcularValores(gente)
         actualizarValores(datos)
```
Declarar una variable como datos/data, es lo mismo que poner "coso", que son esos valores que devuelve la función? el nombre debería ser mas preciso de acuerdo a eso.

gruvix replied:
jajaja "coso"
lo entiendo perfectamente, ya estoy mejorando la sintaxis

GonzaloAlt commented on js/tarea1.js:
```diff
@@ -1,36 +1,52 @@
 }
 
-function validarInputCantidad(input){
+function validarCantidadFamiliares(input){
```
Que es input? si la función se llama `validarCantidadFamiliares`, no debería recibir `cantidad`?

GonzaloAlt commented on js/tarea1.js:
```diff
@@ -1,36 +1,52 @@
 
-function validarInputCantidad(input){
+function validarCantidadFamiliares(input){
     if(input < 1){
```
Evitemos los `magic numbers` y guardemos los números en variables porfa.

GonzaloAlt commented on js/tarea1.js:
```diff
@@ -40,7 +56,6 @@ function validarInputCantidad(input){
     if(input % 1 != 0){
```
Lo mismo acá.

GonzaloAlt commented on js/tarea1.js:
```diff
@@ -65,7 +80,8 @@ function agregarInputs(){
 function validarEdades(gente){
```
En esta función, estas validando las edades o estas manejando como mostrás los errores?

GonzaloAlt commented on js/tarea1.js:
```diff
@@ -65,7 +80,8 @@ function agregarInputs(){
 function validarEdades(gente){
     let noHayErrores = true;
```
`let error` tal vez?

GonzaloAlt commented on js/tarea1.js:
```diff
@@ -65,7 +80,8 @@ function agregarInputs(){
+        const edad = Number(gente[index].value)
+        const errores = validarEdad(edad)
 
         switch(errores){
```
Las dos condiciones en el switch, no hacen lo mismo?

gruvix replied:
no entiendo
el Switch se encarga de manejar cada tipo de error (o de mostrar las cosas bien en caso de q no haya error)
pero el switch necesita antes recibir un error, el cual se lo da validarEdad

GonzaloAlt commented on js/tarea1.js:
```diff
@@ -86,6 +102,9 @@ function validarEdades(gente){
         ocultarErroresEdades()
     }
+    else{
+        mostrarErroresEdades()
```
Tal vez `mostrarErroresEdades()` debería recibir los errores a mostrar?

gruvix replied:
esto lo hace la funcion -mal nombrada- validarEdades (que ya renombré en el futuro pull)

gruvix committed:
mejora de sintaxis

cambia el nonmbre de varias funciones y variables para mejor comprension del script
no se modifica la funcionalidad

gruvix committed:
renombra funcion validarEdad

renombra la funcion validarEdades por un titulo mas correcto a lo que cumple la funcion

gruvix committed:
renombra funcion a plural

la funcion manejarErroresEdades maneja múltiples errores

gruvix committed:
change magic numbers validaciones

ahora se usan constantes para las validaciones de la cantidad y las edades

gruvix committed:
renombra funcion de reiniciado

gruvix committed:
fix ya no se muestran errores viejos junto a los nuevos

los errores de edades ya no permanecen junto a nuevos errores, primero se limpian y luego se vuelven a crear

gruvix committed:
improve manejarErroresEdades

se inviertió la funcionalidad de manejarErroresEdades

gruvix committed:
ocultar resultados si hay errores

se ocultan los resultados viejos de maximo, minimo y promedio si hay errores de ingreso

gruvix committed:
limpieza de espacios innecesarios

gruvix committed:
mejora de sintaxis de manejarErroresEdades

ahora se declara una constante antes de comparar manejarErroresEdades
se renombra la variable hayErrores a simplemente "error"
se renombra gente a inputsEdades, ya que explica mejor lo que lleva la variable
